ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.stories.tsx

```diff
@@ -0,0 +1,63 @@
+import type { Meta, StoryObj } from '@storybook/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+import mockEstimates from '../../../../../../test/data/mock-estimates.json';
+import mockState from '../../../../../../test/data/mock-state.json';
+import configureStore from '../../../../../store/store';
+import { AdvancedGasFeePopoverContextProvider } from '../context';
+import { GasFeeContextProvider } from '../../../../../contexts/gasFee';
+import { GasEstimateTypes } from '../../../../../../shared/constants/gas';
+import { getSelectedInternalAccountFromMockState } from '../../../../../../test/jest/mocks';
+import AdvancedGasFeeGasLimit from './advanced-gas-fee-gas-limit';
+
+const mockInternalAccount = getSelectedInternalAccountFromMockState(mockState);
+
+const store = configureStore({
+  ...mockState,
+  metamask: {
+    ...mockState.metamask,
+    accounts: {
+      [mockInternalAccount.address]: {
+        address: mockInternalAccount.address,
+        balance: '0x1F4',
+      },
+    },
+    featureFlags: { advancedInlineGas: true },
+    gasFeeEstimates: mockEstimates[GasEstimateTypes.feeMarket].gasFeeEstimates,
+  },
+});
+
+const meta: Meta<typeof AdvancedGasFeeGasLimit> = {
+  title: 'Pages/Confirmations/Components/AdvancedGasFeePopover/AdvancedGasFeeGasLimit',
+  component: AdvancedGasFeeGasLimit,
+  decorators: [
+    (Story) => (
+      <Provider store={store}>
+        <GasFeeContextProvider
+          transaction={{
+            userFeeLevel: 'custom',
+            txParams: { gas: '0x5208' }, // 21000 in hex
+            chainId: '0x1',
+            originalGasEstimate: '0x5208',
+          }}
+        >
+          <AdvancedGasFeePopoverContextProvider>
+              <Story />
+          </AdvancedGasFeePopoverContextProvider>
+        </GasFeeContextProvider>
+      </Provider>
+    ),
+  ],
+  parameters: {
+    docs: {
+      description: {
+        component: 'Component for displaying and editing gas limit in advanced gas fee popover',
+      },
+    },
+  },
+};
+
+export default meta;
+type Story = StoryObj<typeof AdvancedGasFeeGasLimit>;
+
+export const Default: Story = {};
```

Comment:
Adds story for `ui/pages/confirmations/components/advanced-gas-fee-popover/advanced-gas-fee-gas-limit/advanced-gas-fee-gas-limit.js` component

<img width="532" alt="Screenshot 2025-06-23 at 2 20 25 PM" src="https://github.com/user-attachments/assets/00c3ac50-2d5d-4e26-ba2e-9ab26a853428" />